pkgs/development/tools/ocaml/js_of_ocaml/compiler.nix

```diff
@@ -1,28 +1,28 @@
 { lib, fetchurl, buildDunePackage
-, ocaml, findlib, cmdliner, dune_2, cppo, yojson, ocaml-migrate-parsetree
+, cmdliner, cppo, yojson, ppxlib
 , menhir
 }:
 
 buildDunePackage rec {
-	pname = "js_of_ocaml-compiler";
-	version = "3.7.1";
-	useDune2 = true;
+  pname = "js_of_ocaml-compiler";
+  version = "3.8.0";
+  useDune2 = true;
 
-	src = fetchurl {
-		url = "https://github.com/ocsigen/js_of_ocaml/releases/download/${version}/js_of_ocaml-${version}.tbz";
-		sha256 = "0i0smhvsfx2ydmbyg5ai5cgqsfnng8rkcvys4i3fa55cw24aknrn";
-	};
+  src = fetchurl {
+    url = "https://github.com/ocsigen/js_of_ocaml/releases/download/${version}/js_of_ocaml-${version}.tbz";
+    sha256 = "069jyiayxcgwnips3adxb3d53mzd4rrq2783b9fgmsiyzm545lcy";
+  };
 
-	nativeBuildInputs = [ ocaml findlib dune_2 cppo menhir ];
```

Comment:
These three elements have been put in `nativeBuildInputs` in that PR: #73820. Probably for a reason. Isn’t this reason still valid?

ping @balsoft

Reply:
The reason is obvious -- all of those are `nativeBuildInputs` (they are build-time dependencies, so have to run on `build` platform) and not `buildInputs` (which would run on `host` platform).

Reply:
I was a bit too eager when deleting stuff here it seems. According to [js_of_ocaml-compiler's opam file](https://github.com/ocsigen/js_of_ocaml/blob/3.8.0/js_of_ocaml-compiler.opam) cppo is no longer required, but I'll re-add menhir.

Reply:
Nevermind what I wrote earlier, I wouldn't see why this made any difference or they would be required as `buildDunePackage` adds them to `buildInputs`. I think we don't support cross compiling for OCaml packages so this is no issue per se?!

Reply:
Thanks. But my question was more: is there a realistic scenario in which the host and build platforms are different?

Reply:
I see. I'd say it'd be simpler to address that in buildDunePackage for the whole of `ocamlPackages`.